frame/support/test/tests/construct_runtime_ui/undefined_validate_unsigned_part.rs

```diff
@@ -52,8 +52,8 @@ construct_runtime! {
 		NodeBlock = Block,
 		UncheckedExtrinsic = UncheckedExtrinsic
 	{
-		System: frame_system::{Pallet, Call, Storage, Config, Event<T>},
-		Pallet: pallet::{Pallet, ValidateUnsigned},
+		System: frame_system expanded::{}::{Pallet, Call, Storage, Config, Event<T>},
```

Comment:
I've reverted this to remove the `expanded` word, the output looks similar to before.
Some small differences in how the compiler reports the error, most probably related to the extra expansion

### Example

 [missing_event_generic_on_module_with_instance.rs](https://github.com/paritytech/substrate/blob/34134935dd894d8322c1de72e95cc5d37780ff32/frame/support/test/tests/construct_runtime_ui/missing_event_generic_on_module_with_instance.rs#L1C1-L14) (this is the entire file)

```rust
use frame_support::construct_runtime;

construct_runtime! {
	pub struct Runtime where
		Block = Block,
		NodeBlock = Block,
		UncheckedExtrinsic = UncheckedExtrinsic
	{
		System: system::{Pallet},
		Balance: balances::<Instance1>::{Event},
	}
}

fn main() {}
```

If we don't add the `expanded` keyword this would produce:

```
error[E0433]: failed to resolve: use of undeclared crate or module `balances`
  --> tests/construct_runtime_ui/missing_event_generic_on_module_with_instance.rs:10:12
   |
10 |         Balance: balances::<Instance1> ::{Event},
   |                  ^^^^^^^^ use of undeclared crate or module `balances`
```

When we would actually expect:

```
error: Instantiable pallet with no generic `Event` cannot be constructed: pallet `Balance` must have generic `Event`
  --> $DIR/missing_event_generic_on_module_with_instance.rs:10:3
   |
10 |         Balance: balances::<Instance1> ::{Event},
   |         ^^^^^^^
```

I'm presuming that because the `construct_runtime!` is not in it's final expression (does not contain the expanded), the macro tries to expand the pallet one more time. In doing so it calls `balances::tt_extra_parts!`, which does not exist (ie theres no `balances` module).

The "Instantiable pallet with no generic `Event` cannot be constructed: pallet `Balance` must have generic" compile error should be coming from the final step of the construct_runtime! macro. And I think it's not outputted/detected because the missing module error has priority for the compiler 🤔 

Let me know if this sounds right :D

Reply:
@bkchr would love to get your feedback on the above 🙏